infra/cloudformation/metric-stream.yaml

```diff
@@ -0,0 +1,124 @@
+
+    BackupBucket:
+        Type: AWS::S3::Bucket
+        DeletionPolicy: Delete
+        UpdateReplacePolicy: Delete
+        Properties:
+            LifecycleConfiguration:
+                Rules:
+                    - Id: DeleteOldBackups
+                      Status: Enabled
+                      ExpirationInDays: 14
+
+    FirehoseRole:
+        Type: AWS::IAM::Role
+        Properties:
+            AssumeRolePolicyDocument:
+                Version: '2012-10-17'
+                Statement:
+                    - Effect: Allow
+                      Principal:
+                          Service: firehose.amazonaws.com
+                      Action: sts:AssumeRole
+            Policies:
+                - PolicyName: FirehosePolicy
+                  PolicyDocument:
+                      Version: '2012-10-17'
+                      Statement:
+                          - Effect: Allow
+                            Action:
+                                - logs:CreateLogStream
+                                - logs:PutLogEvents
+                                - logs:CreateLogGroup
+                            Resource: !Sub arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/kinesisfirehose/${AWS::StackName}:*
+                          - Effect: Allow
+                            Action:
+                                - s3:PutObject
+                                - s3:GetObject
+                            Resource:
+                                - !Sub ${BackupBucket.Arn}/*
+
+    DeliveryStream:
+        Type: AWS::KinesisFirehose::DeliveryStream
+        Properties:
+            DeliveryStreamType: DirectPut
+            HttpEndpointDestinationConfiguration:
+                EndpointConfiguration:
+                    Url: !Ref DestinationEndpoint
+                RequestConfiguration:
+                    ContentEncoding: GZIP
+                    CommonAttributes:
+                        - AttributeName: x-highlight-project
+                          AttributeValue: !Ref HighlightProjectID
+                BufferingHints:
+                    IntervalInSeconds: 60
+                    SizeInMBs: 1
+                RoleARN: !GetAtt FirehoseRole.Arn
+                S3BackupMode: FailedDataOnly
+                S3Configuration:
+                    BucketARN: !GetAtt BackupBucket.Arn
+                    BufferingHints:
+                        IntervalInSeconds: 300
+                        SizeInMBs: 5
+                    CompressionFormat: GZIP
+                    RoleARN: !GetAtt FirehoseRole.Arn
+                    Prefix: failed-metrics/
+
+    MetricStream:
+        Type: AWS::CloudWatch::MetricStream
+        Properties:
+            FirehoseArn: !GetAtt DeliveryStream.Arn
+            IncludeFilters:
+                # Someday we may make this configurable
+                - Namespace: AWS/EC2
+                - Namespace: AWS/Lambda
+                - Namespace: AWS/RDS
```

Comment:
could the default be to send all metrics? in the future, we can allow filtering this down

Reply:
Yeah, that sounds good! I'll update to include all for now.